### PR TITLE
Use go mod instead of dep

### PR DESCRIPTION
**Pull request by boekkooi-fresh**
In order to allow better native support and not require a extra dependency manager it's time to switch to go modules.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,11 +29,8 @@ matrix:
   allow_failures:
     - go: master
 
-before_install:
-  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-
 install:
-  - dep ensure -v
+  - go mod vendor
```
Comment (sergiughf) [inline .travis.yml]:
```suggestion
  - go mod vendor -v
```

Do we want to enable verbose mode?:)

File: .travis.yml [hunk 2]
```diff
@@ -8,19 +8,19 @@ addons:
...
     - POSTGRES_DSN="postgres://postgres@localhost:5432/goengine?sslmode=disable&client_encoding=UTF8"
+    - GO111MODULE=on
 
 matrix:
   include:
     - name: "GoLang CI lint"
       go: 1.11.x
       install:
-        - dep ensure -v
+        - go mod vendor
```
Comment (sergiughf) [inline .travis.yml]:
```suggestion
        - go mod vendor -v
```

Do we want to enable verbose mode?:)

File: Makefile [hunk 3]
```diff
@@ -30,7 +30,7 @@ clean:
 
 deps:
 	$(call title, "Installing dependencies")
-	dep ensure -v
+	go mod vendor
```
Comment (sergiughf) [inline Makefile]:
```suggestion
	go mod vendor -v
```

Do we want to enable verbose mode?:)

**Comment by sergiughf** [comment]
+1

**Comment by cwygoda** [comment]
+1

**Comment by boekkooi-fresh** [comment]
I'm merging without verbose mode because I don't see how it adds value since errors will be reported even without

**Pull request merged**